packages/contracts/scripts/one-off/add-unregistered-contracts.ts

```
@@ -0,0 +1,206 @@
+  } else {
+    const deployerWallet = getDeployerWallet();
+
+    if (!deployerWallet) {
+      throw new Error("Deployer wallet not found");
+    }
+
+    signer = deployerWallet.connect(ethers.provider);
+  }
+
+  let gnosisSetup: {
+    ethAdapter: EthersAdapter;
+    safeApiKit: SafeApiKit;
+    protocolKit: Safe;
+  } | null = null;
+
+  // Gnosis sdk setup
+  if (config.useGnosisSafe) {
+    const ethAdapter = new EthersAdapter({
+      ethers,
+      signerOrProvider: signer,
+    });
+
+    const safeApiKit = new SafeApiKit({
+      txServiceUrl: config.transactionServiceUrl,
+      ethAdapter,
+    });
+
+    const protocolKit = await Safe.create({
+      ethAdapter,
+      safeAddress: config.safeAddress,
+    });
+
+    gnosisSetup = {
+      ethAdapter,
+      safeApiKit,
+      protocolKit,
+    };
+  }
+
+  //////////////////////////////////////////////////////////////////////////////
+  // ACTION BEGINS HERE
+  //////////////////////////////////////////////////////////////////////////////
+
+  // Query for unregistered contracts
+  const registryAddresses = [
+    "0x2ee7b9bb2e038be7323a119701a191c030a61ec6",
+    "0x652490c8bb6e7ec3fd798537d2f348d7904bbbc2",
+  ];
+
+  const client = getClient();
+  const res = await client
+    .query(GetUnregisteredContractsDocument, {
+      registryAddresses,
+    })
+    .toPromise();
+
+  if (!res.data?.contracts_metadata || !res.data.contracts_metadata.length) {
+    console.log("No unregistered contracts found");
+    return;
+  }
+
+  const coreRegistry = CoreRegistryV1__factory.connect(
+    config.coreRegistryAddress,
+    signer
+  );
+
+  const txData: MetaTransactionData[] = [];
+  for (const contract of res.data.contracts_metadata) {
+    const contractCoreVersion = contract.core_version
+      ? ethers.utils.formatBytes32String(contract.core_version)
+      : ethers.constants.HashZero; // "0x0000000000000000000000000000000000000000000000000000000000000000"
```

thould we throw here (and on contractType) instead of falling back to hashZero?

---

Skimming through the result of the query it looks like there are quite a few `GenArt721CoreV2_PBAB` contracts that have a null core_version field, I think throwing on a null contractType makes sense but we'd still need to handle the NULL core_version contracts 🤔 

Example -

```
        "contract_type": "GenArt721CoreV2_PBAB",
        "core_version": null,
        "registered_on": null,
        "address": "0xff124d975c7792e706552b18ec9da24781751cab"
```

---

Every v0-v2 contract will have a null version. Should we just hard code versions for those contract types?

---

Good call - updated this to set the version depending on the contract_type